packages/fluentui/react-northstar-prototypes/src/prototypes/TreeAdditionalActionsPrototype/components/TreeActionsUsingPopup.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import { Tree, Popup, Button, Flex, Text, treeAsListboxBehavior } from '@fluentui/react-northstar';
+
+const TreeActionsUsingPopup = ({ useMacAccessabilityBehavior }) => {
+  const items = [
+    {
+      id: 'tree-item-0',
+      title: 'House Targaryen',
+      items: [
+        {
+          id: 'tree-item-01',
+          title: 'Aerys',
+          items: [
+            {
+              id: 'tree-item-011',
+              title: 'Rhaegar',
+            },
+            {
+              id: 'tree-item-012',
+              title: 'Viserys',
+            },
+            {
+              id: 'tree-item-013',
+              title: 'Daenerys',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      id: 'tree-item-1',
+      title: {
+        children: (C, p) => {
+          return <TreeTitleWithPopupButton />;
+        },
+      },
+      items: [],
+    },
+    {
+      id: 'tree-item-2',
+      title: 'House Targaryen 2',
+      items: [
+        {
+          id: 'tree-item-21',
+          title: 'Aerys',
+          items: [
+            {
+              id: 'tree-item-211',
+              title: 'Rhaegar',
+            },
+            {
+              id: 'tree-item-212',
+              title: 'Viserys',
+            },
+            {
+              id: 'tree-item-213',
+              title: 'Daenerys',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      id: 'tree-item-3',
+      title: 'House Targaryen 3',
+      items: [
+        {
+          id: 'tree-item-31',
+          title: 'Aerys',
+          items: [
+            {
+              id: 'tree-item-311',
+              title: 'Rhaegar',
+            },
+            {
+              id: 'tree-item-312',
+              title: 'Viserys',
+            },
+            {
+              id: 'tree-item-313',
+              title: 'Daenerys',
+            },
+          ],
+        },
+      ],
+    },
+  ];
+
+  const TreeTitleWithPopupButton = () => {
+    const [popupOpen, setPopupOpen] = React.useState(false);
+    const wrapperRef = React.useRef(null);
+
+    return (
+      <Flex
+        role={useMacAccessabilityBehavior ? 'option' : 'treeitem'}
+        data-is-focusable={true}
+        onKeyDown={e => {
+          if (e.key === 'ArrowRight') {
+            e.preventDefault();
+            e.stopPropagation();
+            setPopupOpen(true);
+          }
+        }}
+        fill
+        vAlign="center"
+      >
+        <Flex.Item size={'size.quarter'}>
+          <Text content={'Tree item title'} size="small" />
+        </Flex.Item>
+        <Popup
+          position="above"
+          align="start"
+          offset={({ popper }) => [0, -popper.height]}
+          trapFocus
+          content={
+            <Button
```

Comment:
would be nice to style this button to be the same as the button it is covering, I think there is some font-size/weight difference currently ?